### DISCUSSION TITLE
vtsls: cannot find provider for this feature

### DISCUSSION THREAD

**Discussion by max-svidlo-gett** [LazyVim/LazyVim#3696] [Q&A]
Hi,
First of all, I'm not sure how bad is my skill issues but I'm trying to set up Typescript using LazyExtra (worked fine before moving to vtsls).

I installed vtsls by itself but I keep getting the same error message over and over again:
![image](https://github.com/LazyVim/LazyVim/assets/162127956/81ee1d98-643f-4efe-bfa7-42adf389ad67)

Im using a MacOS

**Comment by Aidan-Chelig** [comment]
having the same issue, im on a mac

**Reply by maxweiner02** [reply]
having the same issue, on windows however

**Comment by folke** [comment]
That's likely because of a tsserver plugin that wasn't found. Like vue

**Reply by amirkarimi** [reply]
I'm working on a [CDK](https://github.com/aws/aws-cdk) project which is pure Typescript (no Vue or React) and getting the same error.

**Comment by GopalVerma1303** [comment]
### Issue: 
Facing the same issue while working on a js/ts project. One thing I observed(just like @amirkarimi) that this error is arising only for pure js/ts projects. There was no error while I was working on a js framework. 
<br />
### Sreenshot:
<img width="1440" alt="Screenshot 2024-07-01 at 1 37 23 PM" 
src="https://github.com/LazyVim/LazyVim/assets/67120469/f3d1ad88-fd7b-474a-b3f5-9ed2ad3f9cee">

**Comment by madflow** [comment]
I think this is related to rather old Typescript versions.

I have created a simple project with just one ".ts" file:

```json
{
  "dependencies": {
    "typescript": "^4.3.2"
  }
}
```

```
   Error  18:57:20 notify.error vtsls: -32603: Cannot find provider for the feature
```

When I update to 

```json
  "dependencies": {
    "typescript": "^4.9.5"
  }
```

The error disappears.

**Comment by max-svidlo-gett** [comment]
Uninstalling `tsserver` and installing `vtsls` again solves this issue for me.

**Reply by madflow** [reply]
> Uninstalling `tsserver` and installing `vtsls` again solves this issue for me.

Probably, because it has been fixed in 0.2.5 ...

https://github.com/yioneko/vtsls/pull/178
https://github.com/yioneko/vtsls/commit/9dff5e3ee5268720648033d661994ac4d160c0cf
